src/records.jl

```diff
@@ -78,7 +78,6 @@ Calling `getproperty` or iterating will evaluate and cache the properties access
 """
 abstract type AttributeRecord <: Record end
 
-@deprecate getindex(rec::T, attr::Symbol) where {T <: Record} getproperty(rec, attr)
```

Comment:
Don't remove deprecations on patch releases.

Reply:
This is also an issue on 0.12.1, but I'm guessing the dataframes release was restricted?

Reply:
> Don't remove deprecations on patch releases.

I agree but I think this is a case of you forgetting to remove this deprecation. Since this is now is only accessible via `Memento.getindex` or `using Memento: getindex` the danger here seems minimal

Reply:
Would you consider doing `@deprecate getindex(...) false` ok?

Reply:
> This is also an issue on 0.12.1, but I'm guessing the dataframes release was restricted?

I think there may be some confusion as to what is going on here. The issue has nothing to do with DataFrames. The issue in particular was created [here](https://github.com/invenia/Memento.jl/commit/44cd87a4a4f3e6ec2c88cadbd3d3ae0a9b1397dc#diff-74d7f3cb64eee8b5198817d9027172f7L14) when you removed `import Base: getindex`. By doing that the `@deprecate getindex(...)` now created a new function called `getindex` which was exported by deprecation.

We can solve this in a few ways:
1. Avoid exporting this alternate `getindex` function
2. Remove the deprecation entirely. This is breaking but this function is already only accessible via `Memento.getindex` or `using Memento: getindex`
3. Update the deprecation to extend `Base.getindex` again

Reply:
I'd vote for 3

Reply:
Out of curiosity why not 1?

Reply:
Does overloading work correctly if it isn't imported though?